src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Check.java

```diff
@@ -5183,6 +5213,12 @@ public Void visitTypeAsEnum(TypeElement e,
                                         TreeInfo.diagnosticPositionFor(method, tree),
                                         Warnings.IneffectualSerialMethodEnum(name));
                         }
+
+                        if (isExtern && externMethodNames.contains(name)) {
+                            log.warning(LintCategory.SERIAL,
```

Comment:
The check for an enum class seems less specific than for record classes, where not only the name but also the flags, return and parameter types are checked.
But for enum classes this also holds for other methods and fields, so it might be a deliberate choice.

Reply:
Good catch; addressed in subsequent push.

As follow-up work, I've filed JDK-8312415: "Expand -Xlint:serial checks to enum constants with specialized class bodies."